src/lib.rs

```diff
@@ -285,6 +288,25 @@ impl WalkDir {
         self.opts.max_open = n;
         self
     }
+
+    /// Set a compare function.
+    /// If a compare function is set, WalkDir will return all paths in sorted
+    /// order.
```

Comment:
Could you fill out this documentation a bit more? In particular, what values are passed to the closure?